homeassistant/components/mqtt/sensor.py

```diff
@@ -262,20 +262,28 @@ def _update_state(msg):
                     self.hass, self._value_is_expired, expiration_at
                 )
 
-            payload = self._template(msg.payload, default=self.native_value)
-
-            if payload is not None and self.device_class in (
+            payload = self._template(msg.payload, default=PayloadSentinel.DEFAULT)
+            if payload is PayloadSentinel.DEFAULT:
+                return
+            if payload is None:
+                self._attr_native_value = None
```

Comment:
L269-270 are untested. When will the template return `None`?

Reply:
This was based on the previous code: `if payload is not None and self.device_class in`
I have taken it out now.